source/module_elecstate/module_dm/test/test_dm_R_init.cpp

```diff
@@ -276,6 +277,38 @@ TEST_F(DMTest, DMInit4)
     delete kv;
 }
 
+// test for save_DMR
+TEST_F(DMTest, saveDMR)
+{
+    // initalize a kvectors
+    K_Vectors* kv = nullptr;
+    int nspin = 2;
+    int nks = 4; // since nspin = 2
+    kv = new K_Vectors;
+    kv->nks = nks;
+    kv->kvec_d.resize(nks);
+    kv->kvec_d[1].x = 0.5;
+    kv->kvec_d[3].x = 0.5;
+    // construct a DM
+    elecstate::DensityMatrix<std::complex<double>, double> DM(kv, paraV, nspin);
+    Grid_Driver gd(0, 0, 0);
+    DM.init_DMR(&gd, &ucell);
+    // construct another DM
+    elecstate::DensityMatrix<std::complex<double>, double> DM_test(kv, paraV, nspin);
+    DM_test.init_DMR(*DM.get_DMR_pointer(1));
```

Comment:
I just leave comment about the function `get_*_pointer` here but not critize your unittest: I would say most of time the function to directly get raw pointer should not be called and the class which owns the variable that the pointer points to should have another design for accessing its variables, even if it is an array.
Get pointer-like operations should only be used in some functions like lapack, ...

Reply:
Got it! Thanks for the information.

Reply:
> I just leave comment about the function `get_*_pointer` here but not critize your unittest: I would say most of time the function to directly get raw pointer should not be called and the class which owns the variable that the pointer points to should have another design for accessing its variables, even if it is an array. Get pointer-like operations should only be used in some functions like lapack, ...

Got it! Thanks for the information.